sdk/plus_actions.go

```diff
@@ -0,0 +1,204 @@
+	client, err := a.extern.ConnectEthClient()
+	if err != nil {
+		return nil, err
+	}
+	defer client.Close()
+
+	plus, err := abigen.NewPlusTransactor(a.queries.Plus(), client)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := plus.MintActivateAndFund(auth, auth.From, cashBackPercent, beneficiary, tier, amount)
+
+	return util.TxPostProcess(tx, err)
+}
+
+func (a *fevmActions) PlusSetPersonalCashBackPercent(ctx context.Context, auth *bind.TransactOpts, tokenID *big.Int, cashBackPercent *big.Int) (*types.Transaction, error) {
+	client, err := a.extern.ConnectEthClient()
+	if err != nil {
+		return nil, err
+	}
+	defer client.Close()
+
+	plus, err := abigen.NewPlusTransactor(a.queries.Plus(), client)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := plus.SetPersonalCashBackPercent(auth, tokenID, cashBackPercent)
+
+	return util.TxPostProcess(tx, err)
+}
+
+func (a *fevmActions) PlusFundGLFVault(ctx context.Context, auth *bind.TransactOpts, tokenID *big.Int, amount *big.Int) (*types.Transaction, error) {
+	client, err := a.extern.ConnectEthClient()
+	if err != nil {
+		return nil, err
+	}
+	defer client.Close()
+
+	plus, err := abigen.NewPlusTransactor(a.queries.Plus(), client)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := plus.FundGlfVault(auth, tokenID, amount)
+
+	return util.TxPostProcess(tx, err)
+}
+
+func (a *fevmActions) PlusClaimCashBack(ctx context.Context, auth *bind.TransactOpts, tokenID *big.Int, receiver common.Address) (*types.Transaction, error) {
+	client, err := a.extern.ConnectEthClient()
+	if err != nil {
+		return nil, err
+	}
+	defer client.Close()
+
+	plus, err := abigen.NewPlusTransactor(a.queries.Plus(), client)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := plus.ClaimCashBack(auth, tokenID, receiver)
+
+	return util.TxPostProcess(tx, err)
+}
+
+func (a *fevmActions) PlusUpgrade(ctx context.Context, auth *bind.TransactOpts, tokenID *big.Int, tier uint8) (*types.Transaction, error) {
+	client, err := a.extern.ConnectEthClient()
+	if err != nil {
+		return nil, err
+	}
+	defer client.Close()
+
+	plus, err := abigen.NewPlusTransactor(a.queries.Plus(), client)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := plus.Upgrade(auth, tokenID, tier)
+
+	return util.TxPostProcess(tx, err)
+}
+
+func (a *fevmActions) PlusDowngrade(ctx context.Context, auth *bind.TransactOpts, tokenID *big.Int, tier uint8, agentAddr common.Address, requesterKey *ecdsa.PrivateKey) (*types.Transaction, error) {
+	client, err := a.extern.ConnectEthClient()
+	if err != nil {
+		return nil, err
+	}
+	defer client.Close()
+
+	closer, err := a.extern.ConnectAdoClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer closer()
+
+	// FIXME: Use downgrade method on ADO
+	jws, err := token.SignJWS(ctx, agentAddr, address.Undef, big.NewInt(0), constants.MethodBorrow, requesterKey, a.queries)
```

Comment:
constants.MethodDowngradePlus could be implemented now - the function signature should not change, can be computed with cast